sklearn/metrics/cluster/_unsupervised.py

```diff
@@ -65,6 +65,7 @@ def silhouette_score(X, labels, *, metric='euclidean', sample_size=None,
 
     labels : array-like of shape (n_samples,)
         Predicted labels for each sample.
+        The labels should be greater than or equal to 0.
```

Comment:
This statement would not be valid for strings:

```py
from sklearn.metrics import silhouette_score
from sklearn.datasets import make_classification

X, _ = make_classification(n_samples=150, random_state=0)
silhouette_score(X, ['a', 'b'] * 75)
# 0.0044188
```

Reply:
So should the statement be modified to include strings ?

Reply:
I think `-1` can be an allowed cluster label.

As for this PR, we can update the docstring such that a user is aware that `-1` is used to denote a noisy sample in the clustering module.

As for removing the noisy label, I am not familiar if one should remove them. Our [example](https://scikit-learn.org/stable/auto_examples/cluster/plot_dbscan.html#sphx-glr-auto-examples-cluster-plot-dbscan-py) does not remove them when calculating the `silhouette_score`.